script/foreman-debug

```diff
@@ -79,50 +79,77 @@ add_cmd() {
-      printv " - $FILE"
-      SUBDIR=$(dirname $FILE)
-      [ ! -d "$DIR$SUBDIR" ] && mkdir -p "$DIR$SUBDIR"
-      MIME=$(file -bi "$FILE" | cut -d\; -f1)
-      case $MIME in
+# add and filter file of known MIME type, calculate size
+add_file() {
+  FILE=$1
+  SIZE=$(du -k $FILE | cut -f1)
+  MIME=$(file -bi "$FILE" | cut -d\; -f1)
+  case $MIME in
         application/x-gzip)
-          zcat "$FILE" | $TAIL_OR_CAT | sed -r "$FILTER" > "$DIR$FILE.txt"
+          zcat "$FILE" | sed -r "$FILTER" > "$DIR$FILE.txt"
           ;;
         application/x-bzip2)
-          bzcat "$FILE" | $TAIL_OR_CAT | sed -r "$FILTER" > "$DIR$FILE.txt"
+          bzcat "$FILE" | sed -r "$FILTER" > "$DIR$FILE.txt"
           ;;
         application/x-xz)
-          xzcat "$FILE" | $TAIL_OR_CAT | sed -r "$FILTER" > "$DIR$FILE.txt"
+          xzcat "$FILE" | sed -r "$FILTER" > "$DIR$FILE.txt"
           ;;
         text/plain | application/xml)
-          $TAIL_OR_CAT "$FILE" | sed -r "$FILTER" > "$DIR$FILE"
+          sed -r "$FILTER" "$FILE" > "$DIR$FILE"
           [ $PRINTPASS -eq 1 ] && grep -H "\*\*\*\*\*" "$DIR$FILE"
           ;;
-        inode/symlink)
-          cp -a "$FILE" "$DIR$FILE"
-          add_files $(readlink -f $FILE)
-          ;;
         *)
           echo "Skipping file $FILE: unknown MIME type $MIME" >> "$DIR/skipped_files"
+          SIZE=0 # don't count the size to collected files
           ;;
       esac
+}
+
+# add files (from newest to oldest) that are non zero, readable, regular file or symlink, until $MAXSIZE limit is reached
+add_files() {
+  SUMSIZE=0
+  SKIP="" # will be nonempty to stop adding files
+  # sort files per modification time, newest first - assuming no space in a filename
+  for FILE in $(find -L $* -printf "%T@ %p\n" 2> /dev/null | sort -nr | cut -d' ' -f2 2> /dev/null); do
+    if [ \( -f "$FILE" -o -h "$FILE" \) -a \( -r "$FILE" -a -s "$FILE" \) ]; then
+      # if we are over size limit, skip rest older files
+      if [ \( $SUMSIZE -gt $MAXSIZE \) -a \( $MAXSIZE -gt 0 \) ]; then
+        printv " - $FILE (skipped due to size)"
+      else
+        printv " - $FILE"
+        SUBDIR=$(dirname $FILE)
+        [ ! -d "$DIR$SUBDIR" ] && mkdir -p "$DIR$SUBDIR"
+        # if the file is symlink, copy source and add target
+        if [ -h $FILE ]; then
+          cp -a "$FILE" "$DIR$FILE"
+          FILE=$(readlink -f $FILE)
+        fi
+        add_file $FILE
+        SUMSIZE=$(($((SUMSIZE))+$((SIZE))))
+      fi
     fi
   done
 }
 
+# call add_files for groups of files that have unique prefix separated by 1st argument
+# example: add_files_types . "/var/log/foreman/*.log*" will call add_files to:
+# /var/log/foreman/apipe_cache.log*
+# /var/log/foreman/cron.log*
+# etc.
+add_files_types() {
+  SEPARATOR="$1"
+  shift
+  FILES="$*"
+  for FILETYPE in $(find -L $FILES 2> /dev/null | awk -F "$SEPARATOR" '{ print $1 }' | sort -u); do
+    add_files "${FILETYPE}${SEPARATOR}*"
+  done
+}
+
```

Comment:
Please remove this function, this is utterly complex. Just use add_files with shell blog for each individual 5MB bundle you want to add. Maybe add a comment below and into katello script but it will be quite obvious and cleaner.